controllers/teleport_master_controller/teleport_master_controller.cpp

```diff
@@ -0,0 +1,169 @@
+        otherRobotsNodes.push_back(tempNode);
+      }
+    
+    // Load config file
+    try {
+        YAML::Node config = YAML::LoadFile("config.yaml");
+        rotations         = config["rotations"].as<std::vector<std::array<double, 4>>>();
+    }
+    catch (const YAML::BadFile& e) {
+        std::cerr << e.msg << std::endl;
+        return 1;
+    }
+    catch (const YAML::ParserException& e) {
+        std::cerr << e.msg << std::endl;
+        return 2;
+    }
+
+    // Open the existing log file and immediately close it to overwrite it for a new instance of
+    // the program
+    std::ofstream log;
+    log.open("teleport_controller.log");
+    log.close();
+    
+    // Get the time step of the current world.
+    int timeStep = int(supervisor.getBasicTimeStep());
+    
+    // Generate random seed
+    std::random_device rd;   // Will be used to obtain a seed for the random number engine
+    std::mt19937 gen(rd());  // Standard mersenne_twister_engine seeded with rd()
+    
+    // Generate distributions for random number systems in the following loop
+    std::uniform_int_distribution<> xDistrib(0, 860);
+    std::uniform_int_distribution<> yDistrib(0, 580);
+    std::uniform_int_distribution<> rotDistrib(0, rotations.size() - 1);
+
+    while (supervisor.step(timeStep) != -1) {
+        
+        // Declare a vector of positions that will be saved as they are randomly generated, to be later
+        // applied to each robot
+        std::vector<std::array<double, 3>> positions = std::vector<std::array<double, 3>>();
+        // Loop through every robot
+        for (long unsigned int i = 0; i < otherRobotsNodes.size(); i++)
+        {
+          // Assume there is no collision
+          bool collision = false;
+          // newPos will be a "Proposed location" for a robot to teleport to
+          std::array<double, 3> newPos;
+          do
+          {
+            collision = false;
+            // Generate a new random location
+            newPos[0] = 4.3 - xDistrib(gen) / 100;
+            newPos[1] = 2.9 - yDistrib(gen) / 100;
+            newPos[2] = 0.51;
+            
+            // Loop through the vector of existing proposed locations and see if the new one is going to
+            // collide with any of them
+            for (std::array<double, 3> testPos : positions)
+            {
+              double distance = sqrt(pow((newPos[1] - testPos[1]), 2) + pow((newPos[0] - testPos[0]), 2));
+              if (distance < 0.1)
+              {
+                collision = true;
+              } 
+            }
+          }
+          // Loop until a proposed location has been found that doesn't clash with the existing ones
+          while (collision); 
+          // Finally add the proposed location in as a confirmed position
+          positions.push_back(newPos);
+        }
+
+
+        // Loop through every robot
+        for (long unsigned int i = 0; i < otherRobotsNodes.size(); i++)
```

Comment:
`long`? Probs wont have that many robots on the field 😄

Reply:
The size of a vector comes in as a long unsigned int, it just comes up with a warning if I have a regular int so that was the easiest way to clear it. I could typecast the size to int if the concern is memory usage

Reply:
It's a vector - is there a reason not to use a range-based `for`?

Reply:
I need the index to apply the relevant position (Line 159)

Reply:
oh true. can you use `size_t` to get rid of the warning here? It's more concise that `long unsigned int` :smile: